sim/core/warrior/warrior.go

```diff
@@ -0,0 +1,170 @@
+package warrior
+
+import (
+	"github.com/wowsims/tbc/sim/core"
+	"github.com/wowsims/tbc/sim/core/proto"
+	"github.com/wowsims/tbc/sim/core/stats"
+)
+
+func RegisterWarrior() {
+	core.RegisterAgentFactory(
+		proto.Player_Warrior{},
+		func(character core.Character, options proto.Player) core.Agent {
+			return NewWarrior(character, options)
+		},
+		func(player *proto.Player, spec interface{}) {
+			playerSpec, ok := spec.(*proto.Player_Warrior)
+			if !ok {
+				panic("Invalid spec value for Warrior!")
+			}
+			player.Spec = playerSpec
+		},
+	)
+}
+
+type Warrior struct {
+	core.Character
+
+	Talents          proto.WarriorTalents
+	Options          proto.Warrior_Options
+	RotationType     proto.Warrior_Rotation_Type
+	ArmsSlamRotation proto.Warrior_Rotation_ArmsSlamRotation
+	ArmsDwRotation   proto.Warrior_Rotation_ArmsDWRotation
+	FuryRotation     proto.Warrior_Rotation_FuryRotation
+}
+
+func (warrior *Warrior) GetCharacter() *core.Character {
+	return &warrior.Character
+}
+
+func (warrior *Warrior) AddPartyBuffs(partyBuffs *proto.PartyBuffs) {
+}
+
+func (warrior *Warrior) Init(sim *core.Simulation) {
+}
+
+func (warrior *Warrior) Reset(newsim *core.Simulation) {
+}
+
+func NewWarrior(character core.Character, options proto.Player) *Warrior {
+	warriorOptions := options.GetWarrior()
+
+	warrior := &Warrior{
+		Character:    character,
+		Talents:      *warriorOptions.Talents,
+		Options:      *warriorOptions.Options,
+		RotationType: warriorOptions.Rotation.Type,
+	}
+
+	warrior.PseudoStats.MeleeSpeedMultiplier = 1
+	warrior.EnableRageBar(warriorOptions.Options.StartingRage)
+	warrior.EnableAutoAttacks(warrior, core.AutoAttackOptions{
+		MainHand:        warrior.WeaponFromMainHand(),
+		OffHand:         warrior.WeaponFromOffHand(),
+		Ranged:          warrior.WeaponFromRanged(),
```

Comment:
I don't think you'll be using ranged attacks in the sim, so you can delete this line. It shouldn't affect results but it might be a small performance improvement, because some calculations will be skipped when there's no ranged weapon.